adapters/restclient/lib/opentelemetry/adapters/restclient/patches/request.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+# Copyright 2020 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Adapters
+    module RestClient
+      module Patches
+        # Module to prepend to RestClient::Request for instrumentation
+        module Request
+          def execute(&block)
+            return super(&block) if disable_span_reporting?
+
+            trace_request do |span|
+              propagate_context(span)
```

Comment:
@mwear It can be moved and the `propagate_context` wrapper method can be replaced. I was only following the style of the Faraday adapter. Would you like me to do your suggested changes?

Reply:
I don't think the extra indirection buys us anything, so I am inclined to make that change. If you agree, and wouldn't mind making the same change to Faraday, that'd be great. I'm open to discussion if you prefer the way things are current structured though.

Reply:
@mwear I agree on removing the indirection.

The only thing that gives me pause about moving the propagation line is that's also where it's done in `dd-trace-rb`. However, I think it would be logically the same to place in `trace_request` before the `yield`.